### DISCUSSION TITLE
Slightly less opinion on the db ?

### DISCUSSION THREAD

**Discussion by rubydesign** [SpinaCMS/Spina#904] [Ideas]
So i am getting started, and i read the bit about spina being opinonated. Which i agree with, in general.

BUT, would it be possible to leave users to choose their db, rather than forcing PG
I really don't like the effort pg puts users through. And while that may mean that i am not a power user to appreciate all the pg benefits, i still wonder. 

Is pg really bringing soo much to the table here? Json attributes are possible in any sql db. Just leave out the `b` , call it `json`

I'd be happy to make the change if the pr would be accepted

**Comment by Bramjetten** [comment]
There are a couple features that Postgres offers that Spina (and some of its plugins) rely on, one of which is jsonb. Some functionality (like querying json paths) can’t be done easily in other databases like MySQL. Another benefit is proper support for full-text search. (used in Spina Shop and Spina Pro)

We could choose to support multiple databases, but that would require a lot more work and maintenance. That’s why we’ve decided to require Postgres for now.

**Comment by rubydesign** [comment]
Thanks for answering. And off course i understand that you use pg and have reasons to.

But now you are forcing that choice on others. And i think you may not use the generator at all?
And the changes are not so big, i'll outline.

So at minimum you could take the hard dependency out. So spina would not be dependent on pg.
And add the dependency as a readme. Because currently your readme (wrongly) says to just add the spina gem. 
Instead you could add spina and pg into gemfile. Or tell people just to replace "jsonb" with "json" after the installer ran

As opposed to either scaring non-pg users (like myself) off.
Or forcing them to maintain a clone and edit that

**Comment by Bramjetten** [comment]
I agree that the pg dependency in the gemspec could be removed. I can't make any promises regarding future compatibility with other databases though. The readme should still say Spina requires Postgres.

**Reply by rubydesign** [reply]
yes it should, because it does not now

If you were pushing a version anyway, maybe . . .
